src/AeroSharp/Connection/ClientProvider.cs

```diff
@@ -24,8 +24,18 @@ internal ClientProvider(ConnectionContext connection, Credentials credentials, C
             };
 
             _hosts = connection.BootstrapServers.Select(s => new Host(s, connection.Port)).ToArray();
-        }
-
+        }
+
+        public Node[] GetNodes()
+        {
+            if (_client is null || !_client.Client.Connected)
+            {
+                _ = GetClient();
+            }
+
+            return _client.GetClientNodes();
+        }

```

Comment:
This doesn't feel like the right abstraction. You've already updated the `ClientWrapper` to include a `GetNodes` method which seems like the correct place to do so.

The Provider's job is to provide you with a client, the ClientWrapper instance should be responsible for doing things with the cluster.

I'd recommend just deleting this outright and removing it from the `IClientProvider` interface.

This would also better align with the description of the interface.

Reply:
oh that indeed sound right , ClientProvider should be only for Client and not its property, their property should be make available from wrapper.